spinalcordtoolbox/scripts/sct_get_centerline.py

```diff
@@ -214,11 +215,21 @@ def main(argv: Sequence[str]):
         return
 
     # Extrapolate and regularize (or detect if optic) cord centerline
-    im_centerline, arr_centerline, _, _ = get_centerline(im_labels,
-                                                         param=param_centerline,
-                                                         space=arguments.space,
-                                                         verbose=verbose,
-                                                         remove_temp_files=arguments.r)
+    im_centerline, _, _, _ = get_centerline(im_labels,
+                                            param=param_centerline,
+                                            # NB: We intentionally skip the pix->phys conversion here, since we want to
+                                            #     convert to phys AFTER the image has been reoriented away from RPI.
+                                            # space=arguments.space,
+                                            verbose=verbose,
+                                            remove_temp_files=arguments.r)
+
+    # Fetch `arr_centerline` from `im_centerline`
+    # (This is the easiest way to get coords that are in their original orientation. This is largely equivalent to how
+    #  arr_centerline is determined inside `get_centerline`, but without the RPI orientation.)
+    arr_centerline = np.array(find_and_sort_coord(im_centerline))
+    # Convert from pixel coordinates to physical coordinates (in original orientation)
+    if arguments.space == "phys":
+        arr_centerline = im_centerline.transfo_pix2phys(arr_centerline.T, mode="absolute").T  # same as get_centerline
```

Comment:
I really suspect that the corresponding lines in `get_centerline` are buggy, if there's a `im_seg.change_orientation` in between generating the voxel coordinates of the centerline and transforming them to physical space. But this will require careful thinking and testing.

Reply:
Your suspicions ended up being 100% correct! :blush: